### ISSUE TITLE
[3.3.5] Core/Pet: HP of pets after dismount

### ISSUE THREAD

**Issue by click** [TrinityCore/TrinityCore#873]
Everytime u dismount, ur pet should have FULL HP no matter what hp it had before you mounted(thats basic rule for all pets).

**Comment by Expecto** [comment]
Any News?

**Comment by Rushor** [comment]
the problem seems to be, that the pet does not get the maxhealth if it has some hp-buffs like blessing-of-kings.

1) summon the pet and give it a hp -buff
2) mount up
3) dismount ( the pet will be resummoned)

-- > the pet only gets the may hp amount without the hp which should be added with the hp buffs.

**Comment by Rushor** [comment]
```c++
diff --git a/src/server/game/Entities/Pet/Pet.cpp b/src/server/game/Entities/Pet/Pet.cpp
index 4c0aeb3..339aff2 100644
--- a/src/server/game/Entities/Pet/Pet.cpp
+++ b/src/server/game/Entities/Pet/Pet.cpp
@@ -262,6 +262,12 @@ bool Pet::LoadPetFromDB(Player* owner, uint32 petEntry, uint32 petnumber, bool c
     SetReactState(ReactStates(fields[6].GetUInt8()));
     SetCanModifyStats(true);
 
+    InitTalentForLevel();                                   // set original talents points before spell loading
+
+    uint32 timediff = uint32(time(NULL) - fields[14].GetUInt32());
+    if (!isInArenaJoinPhase)
+        _LoadAuras(timediff);
+
     if (getPetType() == HUNTER_PET)              
         SetPower(POWER_FOCUS, GetMaxPower(POWER_FOCUS));
 
@@ -320,13 +326,6 @@ bool Pet::LoadPetFromDB(Player* owner, uint32 petEntry, uint32 petnumber, bool c
     }
 
     owner->SetMinion(this, true);

-    InitTalentForLevel();                                   // set original talents points before spell loading
-
-    uint32 timediff = uint32(time(NULL) - fields[14].GetUInt32());
-    if (!isInArenaJoinPhase)
-        _LoadAuras(timediff);
 
     // load action bar, if data broken will fill later by default spells.
     if (!isTemporarySummon)
```

this fixes the issue

**Comment by DevRival** [comment]
https://github.com/TrinityCore/TrinityCore/issues/15030

**Comment by DevRival** [comment]
https://github.com/TrinityCore/TrinityCore/pull/15074

**Comment by Runico** [comment]
@Rushor, that's only works for HP, what about Mana ?

**Comment by Rushor** [comment]
you are shure that mana should be updated as well?

**Comment by Runico** [comment]
@Rushor, No, I'm not shoure

**Comment by Rushor** [comment]
the last fix on this was https://github.com/TrinityCore/TrinityCore/pull/15667 could someone reopen a pr for this?

**Comment by tkrokli** [comment]
Sure thing. Do you mean the fix from #15667
```diff
diff --git a/src/server/game/Spells/SpellEffects.cpp b/src/server/game/Spells/SpellEffects.cpp
index 9535ca2..11cdcea 100644
--- a/src/server/game/Spells/SpellEffects.cpp
+++ b/src/server/game/Spells/SpellEffects.cpp
@@ -3020,6 +3020,13 @@ void Spell::EffectSummonPet(SpellEffIndex effIndex)
             //OldSummon->Relocate(px, py, pz, OldSummon->GetOrientation());
             //OldSummon->SetMap(owner->GetMap());
             //owner->GetMap()->Add(OldSummon->ToCreature());
+            if (OldSummon->getPetType() == SUMMON_PET)
+            {
+                OldSummon->SetHealth(OldSummon->GetMaxHealth());
+                OldSummon->SetPower(POWER_MANA, OldSummon->GetMaxPower(POWER_MANA));
+                OldSummon->SetPower(POWER_ENERGY, OldSummon->GetMaxPower(POWER_ENERGY));
+                OldSummon->GetSpellHistory()->ResetAllCooldowns();
+            }
 
             if (owner->GetTypeId() == TYPEID_PLAYER && OldSummon->isControlled())
                 owner->ToPlayer()->PetSpellInitialize();
```
... or your diff above?

**Comment by Rushor** [comment]
the one from #15667 :)
